webapp/app/.server/service/l2-cancellations.ts

```diff
@@ -234,3 +274,44 @@ export function getL2GovernorAddress(proposalType: L2CancellationType) {
   }
   return l2GovernorAddress;
 }
+
+export async function getAndUpdateL2CancellationByExternalId(
+  externalId: Hex
+): Promise<InferSelectModel<typeof l2CancellationsTable>> {
+  const proposal = await getL2CancellationByExternalId(externalId);
+  if (!proposal) {
+    throw notFound();
+  }
+
+  return await upgradeCancellationStatus(proposal, await getL2VetoNonce());
+}
+
+async function upgradeCancellationStatus(
+  cancellation: InferSelectModel<typeof l2CancellationsTable>,
+  currentNonce: number
+): Promise<InferSelectModel<typeof l2CancellationsTable>> {
+  if (cancellation.status !== l2CancellationStatusEnum.enum.ACTIVE) {
+    return cancellation;
+  }
+
+  if (hexSchema.safeParse(cancellation.transactionHash).success) {
```

Comment:
I believe this `hexSchema` check shouldn't be necessary, as `transactionHash` is already of type `bytea`

Reply:
We agreed on leaving this for next release.